src/backend/BinaryOperator.js

```diff
@@ -9,16 +9,32 @@ const Node = require('./Node.js');
 
 class BinaryOperator extends Node {
 
-  constructor() {
+  constructor(operatorText) {
     super();
     this.leftOperand = null;
     this.rightOperand = null;
-    this.operatorText = '';
+    this.operatorText = operatorText;
     if (this.constructor === BinaryOperator) {
       throw new TypeError('Abstract class \'BinaryOperator\' cannot be instantiated');
     }
   }
 
+  setLeftOperand(leftOperand) {
+    if (leftOperand instanceof Node) {
+      this.leftOperand = leftOperand;
+    } else {
+      this.leftOperand = null;
+    }
+  }
+
+  setRightOperand(rightOperand) {
+    if (rightOperand instanceof Node) {
+      this.rightOperand = rightOperand;
+    } else {
+      this.rightOperand = rightOperand;
+    }
+  }
+
   getText() {
```

Comment:
Talk with @mattfrautnick one of you needs to change this so that your classes agree

Reply:
@mattfrautnick can probably remove the get text() methods from his classes, since it is implemented by the parent class.

Reply:
Sounds good with me.